applications/luci-app-filemanager/htdocs/luci-static/resources/view/system/filemanager.js

```diff
@@ -0,0 +1,2288 @@
+
+            var lineNumbersDiv = document.getElementById('line-numbers');
+            var editorTextarea = document.getElementById('editor-textarea');
+
+            if (lineNumbersDiv && lineNumbersDiv.style.display !== 'none') {
+                self.updateLineNumbers();
+                editorTextarea.addEventListener('input', self.updateLineNumbers.bind(self));
+            }
+
+            // Синхронизируем прокрутку
+            editorTextarea.addEventListener('scroll', self.syncScroll.bind(self));
+            self.syncScroll();
+
+            // Настраиваем нижний отступ для line-numbers
+            self.adjustLineNumbersPadding();
+            window.addEventListener('resize', self.adjustLineNumbersPadding.bind(self));
+
+            var statusInfo = document.getElementById('status-info');
+            if (statusInfo) {
+                statusInfo.textContent = _('Editing: ') + filePath;
+            }
+
+            var statusProgress = document.getElementById('status-progress');
+            if (statusProgress) {
+                statusProgress.innerHTML = '';
+            }
+        }).catch(function(err) {
+            ui.addNotification(null, E('p', _('Failed to open file: %s'.format(err.message))), 'error');
+        });
+    },
+
+    adjustLineNumbersPadding: function() {
+        var lineNumbersDiv = document.getElementById('line-numbers');
+        var editorTextarea = document.getElementById('editor-textarea');
+
+        if (!lineNumbersDiv || !editorTextarea) {
+            return;
+        }
+
+        // Получаем высоту горизонтальной полосы прокрутки
+        var scrollbarHeight = editorTextarea.offsetHeight - editorTextarea.clientHeight;
+
+        // Устанавливаем нижний отступ для line-numbers
+        lineNumbersDiv.style.paddingBottom = scrollbarHeight + 'px';
+    },
+
+
+    handleDownloadFile: function(filePath) {
+        var self = this;
+        fs.read(filePath, {
+            binary: true
+        }).then(function(content) {
+            var blob = new Blob([content], {
+                type: 'application/octet-stream'
+            });
+            var downloadLink = document.createElement('a');
+            downloadLink.href = URL.createObjectURL(blob);
+            downloadLink.download = filePath.split('/').pop();
+            document.body.appendChild(downloadLink);
+            downloadLink.click();
+            document.body.removeChild(downloadLink);
+
+            var statusInfo = document.getElementById('status-info');
+            if (statusInfo) {
+                statusInfo.textContent = _('Downloaded: ') + filePath.split('/').pop();
+            }
+        }).catch(function(err) {
+            ui.addNotification(null, E('p', _('Failed to download file: %s'.format(err.message))), 'error');
+        });
+    },
+    handleDeleteFile: function(filePath) {
+        var self = this;
+        if (confirm(_('Are you sure you want to delete this file or directory?'))) {
```

Comment:
do we know which it is before now? Insert a _('file') i18n string depending on a ternary.